project-base/storefront/utils/cart/useCurrentCart.ts

```diff
@@ -2,30 +2,35 @@ import { useCartQuery } from 'graphql/requests/cart/queries/CartQuery.generated'
 import { TypeListedStoreFragment } from 'graphql/requests/stores/fragments/ListedStoreFragment.generated';
 import { TypeTransportWithAvailablePaymentsAndStoresFragment } from 'graphql/requests/transports/fragments/TransportWithAvailablePaymentsAndStoresFragment.generated';
 import { Maybe } from 'graphql/types';
+import { useEffect, useState } from 'react';
 import { usePersistStore } from 'store/usePersistStore';
 import { CurrentCartType } from 'types/cart';
 import { useIsUserLoggedIn } from 'utils/auth/useIsUserLoggedIn';
-import { isStoreHydrated } from 'utils/store/isStoreHydrated';
 
 export const useCurrentCart = (fromCache = true): CurrentCartType => {
     const isUserLoggedIn = useIsUserLoggedIn();
     const cartUuid = usePersistStore((store) => store.cartUuid);
     const packeteryPickupPoint = usePersistStore((store) => store.packeteryPickupPoint);
 
+    const [isCartHydrated, setIsCartHydrated] = useState(false);
     const isWithCart = isUserLoggedIn || !!cartUuid;
 
+    useEffect(() => {
+        setIsCartHydrated(true);
+    }, []);
```

Comment:
I am not sure what naming you would imagine...? We can think about different naming including "cart" I want to keep since this state is used for cart really. And "hydration" is used because it really means that it can be used internally within this hook otherwise you would need to do that basically in every place where you need to work with the cart and components rendered are different on server and client.